modules/internal/ChapelBase.chpl

```diff
@@ -675,9 +675,11 @@ module ChapelBase {
 
   // Removing the 'eltType' arg results in errors for --baseline
   inline proc _ddata_shift(type eltType, data: _ddata(eltType), shift: integral) {
-    var ret: _ddata(eltType);
-     __primitive("shift_base_pointer", ret, data, shift);
-    return ret;
+    local {
```

Comment:
I guess my preference is to remove the local block and just classify `PRIM_ARRAY_SHIFT_BASE_POINTER` as `FAST_NOT_LOCAL`

We're only going to call `_ddata_shift()` when we init, reindex, slice, or change the rank of an array and I don't think a fast-on will (or should) trigger for any of those cases anyways